### PR TITLE
Add ETL and Mapping options

### PR DESCRIPTION
**Pull request by rafarui**
#### Please provide details about this Pull Request (why the change is being made, what this commit will do):
Some ETL and mapping APIs.

### CODE REVIEW

File: pycarol/connectors.py [hunk 1]
```diff
@@ -112,22 +113,27 @@ def delete_by_name(self, name, force_deletion=True):
...
                 Connector ID
             force_deletion: `bool` default `True`
                 Force the deletion
 
         Returns: None
 
         """
+        if mdm_id is not None:
+            _deprecation_msgs("mdm_id is deprecated and will be removed, use connector_id")
+            connector_id = connector_id if connector_id else mdm_id
```
Comment (gzamboni) [inline pycarol/connectors.py:134]: what if both mdm_id and connector_id are None?

File: pycarol/connectors.py [hunk 2]
```diff
@@ -321,10 +326,7 @@ def get_dm_mappings(self, connector_id=None, connector_name=None, staging_name=N
             url = "v1/connectors/mappings/all"
 
         else:
-            if connector_name:
-                connector_id = self.get_by_name(connector_name)['mdmId']
-            else:
-                assert connector_id
+            connector_id = connector_id if connector_id else self.get_by_name(connector_name)['mdmId']
```
Comment (gzamboni) [inline pycarol/connectors.py:329]: on Delete operation there is a check and a deprecation message. 
I think both should have the same behaviour right?

File: pycarol/connectors.py [hunk 3]
```diff
@@ -362,3 +365,272 @@ def get_dm_mappings(self, connector_id=None, connector_name=None, staging_name=N
...
+                print(f'{count}/{total_hits}', end='\r')
+
+        return template_data
+
+    def _play_pause_mapping(self, kind, entity_mapping_id=None, staging_name=None,
+                            connector_name=None, connector_id=None,
+                            reverse_mapping=False,
+                            process_cds=True, ):
+
+        connector_id = connector_id if connector_id else self.get_by_name(connector_name)['mdmId']
```
Comment (gzamboni) [inline pycarol/connectors.py:454]: on Delete operation there is a check and a deprecation message.
I think both should have the same behaviour right?

File: pycarol/data_models/data_models.py [hunk 4]
```diff
@@ -352,38 +352,40 @@ def _get_dm_export_stats(self):
...
+            dm_id: `str` default None
+                 Data model ID
+            delete_records: `boll` default True
+                True to delete and reprocess, False for copy and reprocess.
+
+        Returns: dict
+            Task Definition.
+        """
+
+        dm_id = dm_id if dm_id else self.get_by_name(dm_name)['mdmId']
```
Comment (gzamboni) [inline pycarol/data_models/data_models.py:371]: on Delete operation there is a check and a deprecation message.
I think both should have the same behaviour right?

File: pycarol/connectors.py (hunk 2 above)
Comment (rafarui) [reply pycarol/connectors.py:334]: checking the APIs it is mandatory connecotr_id in this one.. I will add the check

**Comment by rafarui** [comment]
@gzamboni 
Fixed all your points.

**Pull request merged**